File: Python/BOJ_10814_나이순 정렬.py
import sys

num = int(sys.stdin.readline())
member = []
for i in range(num):
    age, name = sys.stdin.readline().rstrip().split()
    member.append([int(age), name, i + 1])

member.sort(key = lambda x : (x[0], x[2]))

for j in range(num):
    print('%d %s' %(member[j][0], member[j][1]))

File: C++/BOJ_10814_나이순 정렬.cpp
#include <iostream>
#include <string>
#include <algorithm>
#include <vector>

using namespace std;

class information {
    public:
        int age, number;
        string name;
};

bool compare(information a, information b)
{
    if (a.age == b.age) return a.number < b.number;
    return a.age < b.age;
}

int main()
{
    ios::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    int num;
    information info;
    vector<information> arr;

    cin >> num;
    for (int i = 0; i < num; i++) {
        cin >> info.age >> info.name;
        info.number = i;
        arr.emplace_back(info);
    }

    sort(arr.begin(), arr.end(), compare);
    for (int i = 0; i < arr.size(); i++) {
        cout << arr[i].age << ' ' << arr[i].name << '\n';
    }
    
    return 0;
}